redis/src/cluster_client.rs

```diff
@@ -121,6 +121,18 @@ impl ClusterParams {
             async_push_sender: value.async_push_sender,
         })
     }
+    fn apply_config(&mut self, config: cluster::ClusterConfig) {
```

Comment:
nit: add empty line.

also, change to 
```rs
fn with_config(&self, config: cluster::ClusterConfig) -> Self {
  let new = self.clone();
   ....
```
Less room for error, IMO.

Reply:
I think it makes more sense and it matches the builder pattern to take it by value and let the caller be responsible of the cloning or not : 
```rust
fn with_config(mut self, config: cluster::ClusterConfig) -> Self {
```

Are you ok with that? If not, I'll go in your direction.

Reply:
that's fine too. It's specifically `& mut` which I try to avoid :)